get_moochub_courses.php

```diff
@@ -29,190 +32,213 @@
-$records = $DB->get_records('ildmeta_vocabulary');
-$vocabularies = new \stdClass();
-foreach ($records as $vocabulary) {
-    $vocabularies->{$vocabulary->title} = manager::filter_vocabulary_lang($vocabulary, 'de');
-}
-
-// Get list of providers.
-$providers = manager::get_providers('de');
+$allmetarecs = TRUE;
 
-// Create a json entry for every course, that is supposed to be shared.
-foreach ($metarecords as $meta) {
-    // Skip courses that are not supposed to be exported or a course record does not exist.
-    if ($meta->noindexcourse != 0 || !$DB->record_exists('course', array('id' => $meta->courseid))) {
-        continue;
+// Check if the 'idn' parameter exists.
+if (isset($_GET['idn'])) {
+    $allmetarecs = FALSE;
+    $idnumber = $_GET['idn'];
+    $course = $DB->get_record('course', array('idnumber' => $idnumber), '*');
+    if($course) {
+        $database_course_id = $course->id;
+        $metarecords = $DB->get_records('ildmeta', array('courseid' => $database_course_id));
     }
+} 
 
-    $metaentry = [];
-    $metaentry['type'] = 'courses';
-    // Create an ID by adding the moodle course id to the host name of the current moodle site.
-    $metaentry['id'] = parse_url($CFG->wwwroot, PHP_URL_HOST) . $meta->courseid;
-    $metaentry['attributes'] = [];
-    $metaentry['attributes']['name'] = $meta->coursetitle;
-    $metaentry['attributes']['courseCode'] = null;
-    $metaentry['attributes']['courseMode'] = 'MOOC';
-
-    $metaentry['attributes']['url'] = manager::get_external_course_link($meta->courseid);
-
-    if (isset($meta->teasertext) && !empty($meta->teasertext)) {
-        $metaentry['attributes']['description'] = $meta->teasertext;
-    } else {
-        $metaentry['attributes']['description'] = null;
-    }
-    $metaentry['attributes']['abstract'] = null;
-
-    // Set teaching language.
-    // TODO: Enable selection of multiple teaching languages.
-    // TODO: Add more languages and manage them like the other vocabularies.
-    $langlist = [
-        'de',
-        'en',
-        'uk',
-        'ru'
-    ];
-    $metaentry['attributes']['languages'] = [$langlist[$meta->courselanguage]];
-    
-    $metaentry['attributes']['startDate'] = date('c', $meta->starttime);
-    $metaentry['attributes']['endDate'] = null;
-
-    // Get overview image from filestorage.
-    // If no custom image ist set in ildmeta, then use the course image instead.
-    $fs = get_file_storage();
-    $context = context_course::instance($meta->courseid);
-    $fileurl = '';
-    $imagefile = null;
-    if (isset($meta->overviewimage)) {
-        $files = $fs->get_area_files($context->id, 'local_ildmeta', 'overviewimage', 0);
-    } else {
-        $files = $fs->get_area_files($context->id, 'course', 'overviewfiles', 0);
+if ($allmetarecs) {
+    $metarecords = $DB->get_records('ildmeta');
+}
+
+// Return empty data, if there was no course.
+if(!$metarecords) {
+    $metas['data'] = null;
```

Comment:
data darf laut schema nicht null sein
```suggestion
    $metas['data'] = [];
```

Reply:
Alternativ könnten wir hier auch ein 404 Error werfen. Da eine leeres Ergebnis vermutlich in keinem Fall eine erwünschter Output wäre.

```suggestion
if (empty($metarecords)) {
    // Send not found.
    header('HTTP/1.1 404 Not Found, no courses found.');
    die();
}
```

Reply:
Das ist keine günstige Lösung aus meiner Sicht. Bei der automatischen Weiterverabreitung erwarte ich auf Anwenderseite, dass es eine Schleife über die Daten gibt und irgendwas macht. Daher würde ich die leeren data senden.